base/MAPL_MaxMinMod.F90

```diff
@@ -90,25 +87,10 @@ subroutine pmaxmin2d_r4 ( qname, a, pmin_, pmax_, fac_ )
 
       call ESMF_VmGetCurrent(vm=vm, rc=status)
 
-      do j=1,jt
-         pmax = a(1,j)
-         pmin = a(1,j)
-         do i=2,im
-            pmax = max(pmax, a(i,j))
-            pmin = min(pmin, a(i,j))
-         enddo
-         qmax(j) = pmax
-         qmin(j) = pmin
-      enddo
-!
-! Now find max/min of amax/amin
-!
-      pmax = qmax(1)
-      pmin = qmin(1)
-      do j=2,jt
-         pmax = max(pmax, qmax(j))
-         pmin = min(pmin, qmin(j))
-      enddo
+      has_nans = any(a /= a)
```

Comment:
Query for @tclune: What is the best way to figure out if there are NaNs? This? Or should we do:
```fortran
has_nans=any(ieee_is_nan(a))
```

That is, is `any(a /= a)` perhaps more "generic" than using `ieee_is_nan()`?

Reply:
Adopted suggestions with  modifications